### PR TITLE
TST/REF: collect DataFrame reduction tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
NB: this retains all extant fixture usage.

Collect tests by method being tested.  Collect all calls to assert_stat_op_api in one test (one test which can be usefully parametrized/fixturized in a follow-up), removing a ton of noise from other test function signatures.  As a result, it becomes much easier to see places where we can _usefully_ parametrize/fixturize tests in this module.  e.g. test_min and test_max are nearly identical.  ditto test_cumsum/test_cumprod/test_cummin/test_cummax.  ditto test_idxmin/test_idxmax...

### CODE REVIEW

File: pandas/tests/frame/test_analytics.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+f # -*- coding: utf-8 -*-
```
Comment (gfyoung) [inline pandas/tests/frame/test_analytics.py]: Why is there an "f" here?

File: pandas/tests/frame/test_analytics.py [hunk 2]
```diff
@@ -789,10 +812,44 @@ def test_stat_operators_attempt_obj_array(self, method):
             if method in ['sum', 'prod']:
                 tm.assert_series_equal(result, expected)
 
-    def test_mean(self, float_frame_with_na, float_frame, float_string_frame):
-        assert_stat_op_calc('mean', np.mean, float_frame_with_na,
-                            check_dates=True)
-        assert_stat_op_api('mean', float_frame, float_string_frame)
+    @pytest.mark.parametrize('op', ['mean', 'std', 'var',
+                                    'skew', 'kurt', 'sem'])
```
Comment (gfyoung) [inline pandas/tests/frame/test_analytics.py]: I see you decide to add some `pytest` idiom here.  There are a bunch of other places (that you modified) that definitely merit some more refactoring as you probably know...

If you are intending to add more parameterization, then I'll comment as needed.

File: pandas/tests/frame/test_analytics.py [hunk 3]
```diff
@@ -693,82 +696,113 @@ def test_describe_tz_values(self, tz_naive_fixture):
...
+        def mad(x):
+            return np.abs(x - x.mean()).mean()
+
+        def var(x):
+            return np.var(x, ddof=1)
+
+        def std(x):
+            return np.std(x, ddof=1)
+
+        def sem(x):
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: what are all these for?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_analytics.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_analytics.py]: These get passed to assert_stat_op_calc.  In the status quo these are defined in separate test methods and are mostly named `alt`

File: pandas/tests/frame/test_analytics.py [hunk 4]
```diff
@@ -693,82 +696,113 @@ def test_describe_tz_values(self, tz_naive_fixture):
...
 
+        assert_stat_op_calc('sum', np.sum, float_frame_with_na,
+                            skipna_alternative=np.nansum)
+        assert_stat_op_calc('mean', np.mean, float_frame_with_na,
+                            check_dates=True)
+        assert_stat_op_calc('product', np.prod, float_frame_with_na)
+
+        assert_stat_op_calc('mad', mad, float_frame_with_na)
+        assert_stat_op_calc('var', var, float_frame_with_na)
+        assert_stat_op_calc('std', std, float_frame_with_na)
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: in a future pass should parameterize this test

**Comment by jreback** [comment]
looks fine, can you merge master. ping on green.

**Comment by jbrockmendel** [comment]
Ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**